dyc/top.py

```diff
@@ -0,0 +1,140 @@
+import sys
+import re
+import fileinput
+import copy
+import linecache
+import click
+from .utils import (
+    get_leading_whitespace,
+    BlankFormatter,
+    get_indent,
+    add_start_end,
+    get_file_lines,
+)
+from .base import Builder
+
+
+class TopBuilder(Builder):
+    def __init__(self, filename, config, placeholders=False):
+        self.filename = filename
+        self.config = config
+        self.placeholders = placeholders
+
+    def initialize(self):
```

Comment:
We could use the `initialize` for setting up the properties to capture. Also since we're not calling that from `process_top`. Then that won't be needed. 

It's preferable to follow the same workflow of callables here. Just so that could also be refactored to use the same callables for all three sections of docstrings.
```
builder = MethodBuilder(
    filename, method_cnf, placeholders=self.placeholders
)
builder.initialize(change=change)
builder.prompts()
builder.apply()
builder.clear(filename)
```